### PR TITLE
refactor(settings-row): convert enzyme tests to rtl

### PR DESCRIPTION
**Pull request by tomdavies73**
### Proposed behaviour

Converts enzyme tests ro rtl.

### Current behaviour

Tests are currently written in enzyme.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [x] Unit tests added or updated if required
- [ ] Playwright automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [ ] Tested in provided StackBlitz sandbox/Storybook
- [ ] Add new Playwright test coverage if required
- [ ] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### CODE REVIEW

File: src/components/settings-row/settings-row.test.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+);
+
+test("renders with the correct description text when the `description` prop is passed", () => {
+  render(<SettingsRow title="foo" description="bar" />);
+
+  const description = screen.getByText("bar");
+  expect(description).toBeVisible();
+});
+
+test("renders a separator element when the `description` prop is passed", () => {
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.test.tsx:59]: **suggestion/question:** This test assertion doesn't seem to match the description here. It states `renders a separator element` but the assertion is getting the styles applied to the `description`. Is this correct?

File: src/components/settings-row/settings-row.test.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+  render(<SettingsRow title="foo" description="bar" />);
+
+  const description = screen.getByText("bar");
+  expect(description).toHaveStyle({ gridRow: "3", marginTop: "0px" });
+});
+
+test("when the `description` prop is not passed, the separator does not render", () => {
```
Comment (DipperTheDan) [inline src/components/settings-row/settings-row.test.tsx:66]: suggestion/question: Same thing here too. This test assertion doesn't seem to match the description. It states doesn't render a separator element but the assertion is querying if the `subtitle` is in the DOM. Is this correct?

Comment (tomdavies73) [reply src/components/settings-row/settings-row.test.tsx:66]: Yeah, good point, this is the styling we apply conditionally internally on `Heading` when the internal prop `hasSeperator` is true.

What I might do is add a `data-role` locator to the `StyledSeperator` component instead and assert against it not being in the DOM 👍

File: src/components/settings-row/settings-row.test.tsx (hunk 1 above)
Comment (tomdavies73) [reply src/components/settings-row/settings-row.test.tsx:59]: Yeah, good point, this is the styling we apply conditionally internally on `Heading` when the internal prop `hasSeperator` is true.

What I might do is add a `data-role` locator to the `StyledSeperator` component instead and assert against its visibility 👍

**Review by DipperTheDan — APPROVED** [review]

**Pull request merged**